app/Services/User/UserService.php

```diff
@@ -171,4 +172,34 @@ public function changePassword(int $id, array $request): bool
             throw new Exception($e->getMessage());
         }
     }
+
+    public function getUserList(array $request): LengthAwarePaginator
+    {
+        try {
+            return $this->userRepository->getList($request);
+        } catch (Exception $e) {
+            throw new Exception($e->getMessage());
+        }
+    }
+
+    public function toggleBlockUser(int $id): bool
+    {
+        try {
+            $user = $this->userRepository->getUserById($id);
+            if ($user->status != User::STATUS_BLOCKED) {
+                $newStatus = [
+                    'status' => User::STATUS_BLOCKED,
+                    'old_status' => $user->status,
```

Comment:
old status để làm gì

Reply:
old_status để em lưu status của user trước khi bị block, not verified hoặc verified, để khi unblock thì trả về đúng status của user trước khi bị block

Reply:
not verified không phải là dùng trường mail_verify_at để kiểm tra à?
làm loằng ngoằng như này ngoài tốn thời gian ra thì đc cái gì k?